akka-stream/src/main/scala/akka/stream/impl/fusing/ActorGraphInterpreter.scala

```diff
@@ -510,6 +518,20 @@ private[stream] class ActorGraphInterpreter(_initial: GraphInterpreterShell) ext
         false
     }
 
+  //this limits number of messages that can be processed synchronously during one actor receive.
+  private val shortCircuitLimit: Int = _initial.mat.settings.syncProcessingLimit
+  private var shortCircuitBuffer: ByteBuffer[BoundaryEvent] = _
+  private var currentLimit: Int = shortCircuitLimit
```

Comment:
That is OK, but why ByteBuffer? But now I see, you renamed stream.impl.Buffer at L10, but this is super confusing now :). We have FixedSizeBuffer (or called something like that) that is array based and you can use it simply.